vip-plugins/vip-plugins.php

```diff
@@ -0,0 +1,222 @@
+	}
+
+	return $plugins;
+}
+
+/**
+ * Render the featured partner plugins to the plugins screenReaderText
+ * Uses the notice hooks as that is all we have on these pages
+ *
+ * @return null
+ */
+function wpcom_vip_render_vip_featured_plugins() {
+	$screen = get_current_screen();
+
+	if ( ! ( 'plugins' === $screen->id || 'plugins-network' === $screen->id ) ) {
+		return;
+	}
+
+	$plugins = wpcom_vip_fetch_vip_featured_plugins();
+
+	if ( ! $plugins ) {
+		return;
+	}
+
+	?>
+	<div class="featured-plugins notice">
+		<h3><?php _e( 'VIP Featured Plugins', 'vip-dashboard' ); ?></h3>
+		<?php
+		foreach ( $plugins as $key => $plugin ) {
+			?>
+			<div class="plugin">
+				<a class="fp-content" href="<?php echo esc_attr( $plugin->meta->plugin_url ); ?>" target="_blank">
+					<img src="<?php echo esc_attr( $plugin->meta->listing_logo ); ?>" alt="<?php echo esc_attr( $plugin->post_title ); ?>" />
+					<h4><?php echo esc_html( $plugin->post_title ); ?></h4>
+					<p><?php echo esc_html( $plugin->meta->listing_description ); ?></p>
+				</a>
+				<a class="fp-overlay" href="<?php echo esc_attr( $plugin->meta->plugin_url ); ?>" target="_blank">
+					<div class="fp-overlay-inner">
+						<div class="fp-overlay-cell">
+							<span>	
+								<?php _e( 'More Information', 'vip-dashboard' ); ?>
+							</span>
+						</div>
+					</div>
+				</a>
+			</div>
+			<?php
+		}
+		?>
+	</div>
+	<?php
+}
+add_action( 'admin_notices', 'wpcom_vip_render_vip_featured_plugins', 99 );
+add_action( 'network_admin_notices', 'wpcom_vip_render_vip_featured_plugins', 99 );
+
+/**
+ * Returns a filtered list of code activated plugins similar to core plugins Option
+ *
+ * @return array list of filtered plugins
+ */
+function wpcom_vip_get_filtered_loaded_plugins() {
+	$code_plugins = wpcom_vip_get_loaded_plugins();
+	foreach ( $code_plugins as $key => $plugin ) {
+		if ( substr( $plugin, 0, 8 ) === 'plugins/' ) {
+			$code_plugins[ $key ] = preg_replace( '/^(plugins\/)/i', '', $plugin );
+		} else {
+			unset( $code_plugins[ $key ] );
+		}
+	}
+
+	return $code_plugins;
+}
+
+/**
+ * Returns a filtered list of code activated plugins similar to network plugins option
+ *
+ * @return array list of filtered, active plugins
+ */
+function wpcom_vip_get_network_filtered_loaded_plugins() {
+	$code_plugins = wpcom_vip_get_filtered_loaded_plugins();
+	foreach ( $code_plugins as $key => $plugin ) {
+		unset( $code_plugins[ $key ] );
+		$code_plugins[ $plugin ] = filemtime( __FILE__ );
+	}
+
+	return $code_plugins;
+}
+
+/**
+ * Ensure code activated plugins are shown as such on core plugins screens
+ *
+ * @param  array $actions
+ * @param  string $plugin_file
+ * @param  array $plugin_data
+ * @param  string $context
+ * @return array
+ */
+function wpcom_vip_plugin_action_links( $actions, $plugin_file, $plugin_data, $context ) {
+	$screen = get_current_screen();
+	if ( in_array( $plugin_file, wpcom_vip_get_filtered_loaded_plugins(), true ) ) {
```

Comment:
Minor: it's cleaner to move function calls like `wpcom_vip_get_filtered_loaded_plugins()` outside of `if` blocks and assign to a variable.